File: mobile/lib/features/condition_travail/domain/enums/jour_semaine.dart
enum JourSemaine {
  lundi('LUNDI', 'Lundi', DateTime.monday),
  mardi('MARDI', 'Mardi', DateTime.tuesday),
  mercredi('MERCREDI', 'Mercredi', DateTime.wednesday),
  jeudi('JEUDI', 'Jeudi', DateTime.thursday),
  vendredi('VENDREDI', 'Vendredi', DateTime.friday),
  samedi('SAMEDI', 'Samedi', DateTime.saturday),
  dimanche('DIMANCHE', 'Dimanche', DateTime.sunday);

  final String backend;
  final String label;
  final int weekday;

  const JourSemaine(this.backend, this.label, this.weekday);

  static JourSemaine? fromJson(dynamic value) {
    if (value == null) return null;
    final s = value.toString();
    for (final day in JourSemaine.values) {
      if (day.backend == s) return day;
    }
    return null;
  }

  String toJson() => backend;
}

File: backend/src/main/java/com/tmk/vtcmanager/application/domain/programmeTravail/JourSemaine.java
package com.tmk.vtcmanager.application.domain.programmeTravail;

import java.time.DayOfWeek;

public enum JourSemaine {
    LUNDI,
    MARDI,
    MERCREDI,
    JEUDI,
    VENDREDI,
    SAMEDI,
    DIMANCHE;

    public static JourSemaine from(DayOfWeek dayOfWeek) {
        return switch (dayOfWeek) {
            case MONDAY    -> LUNDI;
            case TUESDAY   -> MARDI;
            case WEDNESDAY -> MERCREDI;
            case THURSDAY  -> JEUDI;
            case FRIDAY    -> VENDREDI;
            case SATURDAY  -> SAMEDI;
            case SUNDAY    -> DIMANCHE;
        };
    }
}
